common/component/postgresql/awsIAM/aws.go

```diff
@@ -0,0 +1,206 @@
+package awsiam
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	aws_config "github.com/aws/aws-sdk-go-v2/config"
+	aws_credentials "github.com/aws/aws-sdk-go-v2/credentials"
+	aws_auth "github.com/aws/aws-sdk-go-v2/feature/rds/auth"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	pginterfaces "github.com/dapr/components-contrib/common/component/postgresql/interfaces"
+)
+
+const (
+	createDatabaseTmpl = "CREATE DATABASE %s"
+	deleteDatabaseTmpl = "DROP DATABASE IF EXISTS %s WITH (FORCE)"
+	databaseExistsTmpl = "SELECT EXISTS (SELECT 1 FROM pg_database WHERE datname = $1)"
+	userExistsTmpl     = "SELECT CAST((SELECT 1 FROM pg_roles WHERE rolname = $1) AS BOOLEAN)"
+	createUserTmpl     = "CREATE USER %v"
+	checkRoleTmpl      = "SELECT 1 FROM pg_roles WHERE rolname = $1 AND 'rds_iam' = rolname"
+	grantRoleTmpl      = "GRANT %v TO %v"
+
+	awsRole = "rds_iam"
+)
+
+var (
+	// Define a regular expression to match the 'dbname' parameter in the connection string
+	databaseNameRegex = regexp.MustCompile(`\bdbname=([^ ]+)\b`)
+	userRegex         = regexp.MustCompile(`\buser=([^ ]+)\b`)
+)
+
+type AWSIAM struct {
+	// Ignored by metadata parser because included in built-in authentication profile
+	// access key to use for accessing sqs/sns.
+	AWSAccessKey string `json:"awsAccessKey" mapstructure:"awsAccessKey" mdignore:"true"`
+	// secret key to use for accessing sqs/sns.
+	AWSSecretKey string `json:"awsSecretKey" mapstructure:"awsSecretKey" mdignore:"true"`
+	// aws session token to use.
+	AWSSessionToken string `mapstructure:"awsSessionToken" mdignore:"true"`
+	// aws region in which SNS/SQS should create resources.
+	AWSRegion string `mapstructure:"awsRegion"`
+}
+
+func GetAccessToken(ctx context.Context, pgCfg *pgx.ConnConfig, accessKey, secretKey string) (string, error) {
+	var dbEndpoint = fmt.Sprintf("%s:%d", pgCfg.Host, pgCfg.Port)
+	var authenticationToken string
+
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.Connecting.Go.html
+	// Default to load default config through aws credentials file (~/.aws/credentials)
+	awsCfg, err := aws_config.LoadDefaultConfig(ctx)
+	if err != nil {
+		// otherwise use metadata fields
+		// Check if access key and secret access key are set
+		accessKey := accessKey
+		secretKey := secretKey
+
+		// Validate if access key and secret access key are provided
+		if accessKey == "" || secretKey == "" {
```

Comment:
it should be possible to not provide accessKey and secretKey so the aws sdk picks the credentials from the environment and users can use "workload identity" from AWS